src/main/java/com/voicepin/flow/client/FlowClient.java

```diff
@@ -1,45 +1,179 @@
+    /**
+     * Starts verification process on given Voiceprint with incoming stream. Verification
+     * process starts immediately (i.e. it does not wait for the whole stream to be
+     * uploaded).
+     *
+     * @param verifyRequest
+     * @return result client for getting results
+     * @throws FlowClientException
+     */
+    public VerifyStream verify(VerifyRequest verifyRequest) throws FlowClientException {
+
         VerifyInitRequest initReq = new VerifyInitRequest(verifyRequest.getVoiceprintId());
         Call<VerifyInitResult> initCall = new VerifyInitCall(initReq);
         VerifyInitResult initResult = caller.call(initCall);
 
-        VerifyStreamRequest streamReq = new VerifyStreamRequest(initResult.getSpeechPath(), verifyRequest.getSpeechStream());
-        Call<VerifyStreamResult> streamCall = new VerifyStreamCall(streamReq);
-        VerifyStreamResult streamResult = caller.call(streamCall);
+        return new VerifyStream(caller, initResult, verifyRequest.getSpeechStream());
+    }
+
+    /**
+     * Creates builder for FlowClient.
+     * 
+     * @param baseUrl url to the service eg. http://localhost:8081/voicepin-ti-server/v1/
+     *
+     * @return
+     */
+    public static FlowClientBuilder newBuilder(String baseUrl) {
+        return new FlowClientBuilder(baseUrl);
+    }
+
+    public static final class FlowClientBuilder {
+
+        private final String baseUrl;
+
+        private String username;
+        private String password;
+
+        private CertificateStrategy certificateStrategy = new TrustedCertificateStrategy();
+
+        private FlowClientBuilder(String baseUrl) {
+            this.baseUrl = baseUrl;
+        }
+
+        /**
+         * Allows to connect to the Voicepin Flow server using secure http connection with
+         * given credentials.
+         * 
+         * @param username
+         * @param password
+         *
+         * @return
+         */
+        public FlowClientBuilder withHttps(String username, String password) {
+            if (username == null || username.isEmpty()) {
+                throw new IllegalArgumentException("Username is null or empty");
+            }
+
+            if (password == null || password.isEmpty()) {
+                throw new IllegalArgumentException("Password is null or empty");
+            }
+
+            this.username = username;
+            this.password = password;
+            this.certificateStrategy = new TrustedCertificateStrategy();
+            return this;
+        }
+
+        /**
+         * Allows Flow Client to trust all https certificates. Use with caution and only
+         * during development.
+         * 
+         * @return
+         */
+        public FlowClientBuilder acceptAllCertificates() {
+            this.certificateStrategy = new AnyCertificateStrategy();
+            return this;
+        }
+
+        /**
+         * Allows to use custom self-signed certificate.
```

Comment:
Nie tylko self-signed.

Allows certificate pinning - only certificates in passed key store will be accepted (self-signed included).